jenkins-compose.sh

```diff
@@ -17,19 +31,48 @@ cd $(dirname $(readlink -f $0 || realpath $0))
 # we don't use usual .env filename, because docker-compose uses it
 [[ -f env.local ]] && source env.local
 
+############################################################################
+# Handle deprecated/backward-compatible and default variables
+############################################################################
+
+# warning: always make sure to self-reference the variable before to allow sourcing with overrides
+
+# note: the 'SSH' and 'SLAVE' are purposely swapped below to match the legacy name
+export JENKINS_SLAVE_SSH_PUBKEY=${JENKINS_SLAVE_SSH_PUBKEY:-${JENKINS_SSH_AGENT_PUBKEY}}
+export JENKINS_SSH_SLAVE_PUBKEY=${JENKINS_SSH_SLAVE_PUBKEY:-${JENKINS_SLAVE_SSH_PUBKEY}}
+
+# new variables for convenience/alignment of older 'ssh-slave' naming
+# note: for legacy reasons, the '..._SSH_SLAVE_...' names can also be used
+export JENKINS_SSH_AGENT_VERSION=${JENKINS_SSH_AGENT_VERSION:-${JENKINS_SSH_SLAVE_VERSION}}
+export JENKINS_SSH_AGENT_DOCKER=${JENKINS_SSH_AGENT_DOCKER:-${JENKINS_SSH_SLAVE_DOCKER}}
+export JENKINS_SSH_AGENT_IMAGE=${JENKINS_SSH_AGENT_IMAGE:-${JENKINS_SSH_SLAVE_IMAGE}}
```

Comment:
Wait a sec, all those older `SLAVE` images var never existed before because they were not in the previous `env.local` version !  The images were hardcoded before.